lib/model.js

```diff
@@ -290,7 +291,7 @@ module.exports = (function() {
     this.Instance.prototype.validators = {};
 
     Utils._.each(this.rawAttributes, function(definition, name) {
-      var type = definition.type._typeName || definition.type;
+      if (typeof definition.type === "function") definition.type = new definition.type();
```

Comment:
We have multiple entrypoints for data types though. addColumn, createTable, and models in general